File: PHP/MiniMaxSumTest.php
<?php
/**
 * @version PHP 7.3.29
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit NomeDaClasseTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("MiniMaxSum.php");

class MiniMaxSumTest extends TestCase {

    public function testMiniMaxSum() {
        $obj = new MiniMaxSum();
        //Test 1
        $want = "10 14";
        $got = $obj->miniMaxSum( [1, 2, 3, 4, 5] );
        echo "\nTest 1: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 2
        $want = "16 24";
        $got = $obj->miniMaxSum( [1, 3, 5, 7, 9] );
        echo "\nTest 2: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 3
        $want = "299 9271";
        $got = $obj->miniMaxSum( [7, 69, 2, 221, 8974] );
        echo "\nTest 3: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 4
        $want = "1673711044 2486347135";
        $got = $obj->miniMaxSum( [140638725, 436257910, 953274816, 734065819, 362748590] );
        echo "\nTest 4: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
    }

}

File: GO/MiniMaxSum_test.go
/**
 * @version GO 1.19.2
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import (
	"reflect"
	"testing"
)

func TestMiniMaxSum(t *testing.T) {
	tables := []struct {
		arr  []int32
		want string
	}{
		{[]int32{1, 2, 3, 4, 5}, "10 14"},
		{[]int32{1, 3, 5, 7, 9}, "16 24"},
		{[]int32{7, 69, 2, 221, 8974}, "299 9271"},
		{[]int32{140638725, 436257910, 953274816, 734065819, 362748590}, "1673711044 2486347135"},
	}

	for _, table := range tables {
		got := miniMaxSum(table.arr)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %v but the return was this: %v", table.want, got)
		}
	}
}
